### PR TITLE
SM-528: Fix SM Import / Export Error Messaging

### PR DESCRIPTION
**Pull request by coltonhurst**
## Type of change

```
- [x] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

See the paired server PR here: https://github.com/bitwarden/server/pull/2734

Remove the try catch so we can use the built-in error messaging.

Additionally, hide the trash and settings tab in the side navigation if the user is not an administrator in the organization.

## Screenshots

Admin view:

admin

Non-admin view:

non-admin

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by Hinton — COMMENTED** [review]
We should lock the pages behind an admin check to prevent the user from navigating to them.

File: bitwarden_license/bit-web/src/app/secrets-manager/settings/settings-routing.module.ts [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+import { ProviderPermissionsGuard } from "../../providers/guards/provider-permissions.guard";
+
 import { SecretsManagerExportComponent } from "./porting/sm-export.component";
 import { SecretsManagerImportComponent } from "./porting/sm-import.component";
 
 const routes: Routes = [
   {
     path: "import",
     component: SecretsManagerImportComponent,
+    canActivate: [ProviderPermissionsGuard],
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/settings/settings-routing.module.ts:15]: Should be organization not provider.

Comment (coltonhurst) [reply bitwarden_license/bit-web/src/app/secrets-manager/settings/settings-routing.module.ts]: Updated

File: bitwarden_license/bit-web/src/app/secrets-manager/layout/navigation.component.ts [hunk 2]
```diff
@@ -9,7 +12,29 @@ import { SecretsManagerLogo } from "./secrets-manager-logo";
...
+    this.route.params
+      .pipe(
+        concatMap(async (params) => {
+          return this.organizationService.get(params.organizationId);
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe((org) => {
+        this.organization = org;
+      });
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/layout/navigation.component.ts:31]: You could change `organization$` to be an observable and use the `async` pipe in the template which removes much of the boilerplate.

Comment (coltonhurst) [reply bitwarden_license/bit-web/src/app/secrets-manager/layout/navigation.component.ts]: Refactored

**Review by Hinton — APPROVED** [review]

**Pull request merged**